WalletWasabi.Backend/Controllers/ChaumianCoinJoinController.cs

```diff
@@ -248,7 +251,7 @@ public async Task<IActionResult> PostInputsAsync([FromBody, Required]InputsReque
 
 						var address = (BitcoinWitPubKeyAddress)txout.ScriptPubKey.GetDestinationAddress(Network);
 						// Check if proofs are valid.
-						if (!address.VerifyMessage(blindedOutputs.First(), inputProof.Proof))
+						if (!address.VerifyMessage(blindedOutputScriptsHash, inputProof.Proof))
```

Comment:
It does not have to be like this. The goal is to make sure the inputs are controlled by the user. The first blinded output is ok to be signed.  
I'd prefer that one, because that results in less computation.

Reply:
I know. But I would prefer to have the blinded outputs also signed anyway. That make the whole request impossible to forge/modify.I think the cost is tiny in comparison with the benefits.